lib/oppen/printer.rb

```diff
@@ -138,6 +140,11 @@ def handle_end(token)
         tokens[right] = token
         size[right] = -1
         scan_stack.push right
+        if config&.indent_anchor == Config::IndentAnchor::ON_BEGIN &&
+           (!scan_stack.empty? && right_total - left_total < print_stack.space)
```

Comment:
Maybe we should extract this to a separate function that expresses the intended meaning pf this check?